{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/logos.png\"\n",
        "        width=450\n",
        "        height=80>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "---\n",
        "# ***Citometr√≠a digital y bioinform√°tica aplicada***\n",
        "---"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ql1UwsZeHBlv"
      },
      "source": [
        "# **Secci√≥n 1: Origen de los datos** üåê"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/Data.png\"\n",
        "        width=900\n",
        "        height=200>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "iMGaRwQe8mI0"
      },
      "source": [
        "## **1.1 - Introducci√≥n**\n",
        "\n",
        "En un proyecto determinado, la aplicaci√≥n de estrategias de c√≥mputo depende de la **informaci√≥n disponible del sistema (DATOS)** y de los **objetivos** espec√≠ficos del estudio.\n",
        "\n",
        "El laboratorio humedo es el primer acercamiento que tenemos a la generaci√≥n de datos, mejor conocidos como: **Datos Crudos**. \n",
        "\n",
        "\n",
        "---\n",
        "Datos Crudos: Registros originales tal como se recopilaron en el laboratorio sin ning√∫n procesamiento, filtrado o interpretaci√≥n.\n",
        "\n",
        "---\n",
        "\n",
        "Al trabajar con muestras biol√≥gicas diversas, es com√∫n analizar:\n",
        "\n",
        "\n",
        "> *   ü©∏ Sangre/plasma (biomarcadores)\n",
        "> *   üî¨ Tejidos (biopsias)\n",
        "> *   üß´ Cultivos celulares (hepatocitos, l√≠neas cancerosas)\n",
        "> *   ü¶† Aislamientos microbianos (resistencia antimicrobiana)\n",
        "\n",
        "\n",
        "\n",
        "Despu√©s de generar los datos, estos pasan a la computadora para proceder a su manejo y adaptarlas acorde a las herramientas de uso.\n",
        "\n",
        "Las herramientas a utilizar es seleccionada basada a la pregunta del proyecto **¬øQue me interesa conocer de mis datos?** \n",
        "\n",
        "\n",
        "### Beneficios:\n",
        "\n",
        "\n",
        "1.   Dejar atr√°s los an√°lisis unig√©nicos (qPCR) para adoptar enfoques multi√≥micos.\n",
        "\n",
        "2.   Vincular sus muestras a mecanismos moleculares con impacto en terapias y diagn√≥sticos\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/bioinfprocess.png\"\n",
        "        width=600\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## 1.2 - Datos transcript√≥micos"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/data_trans.jpg\"\n",
        "        width=1000\n",
        "        height=510>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Nos ayudan a analizar los niveles de expresi√≥n g√©nica en diferentes c√©lulas para inferir **tipos celulares** y **funciones**\n",
        "\n",
        "| üî¨ **Citometr√≠a de Flujo** | üß™ **An√°lisis Transcript√≥mico** |\n",
        "|-------------------------|-----------------------------|\n",
        "| Usa anticuerpos y fluorocromos para identificar prote√≠nas de membrana o intracelulares. | Usa secuenciaci√≥n de ARN para medir la expresi√≥n g√©nica. |\n",
        "| Analiza c√©lulas individualmente en tiempo real. | Puede analizar muestras bulk (mezcla de c√©lulas) o datos de single-cell. |\n",
        "| Limitado por el n√∫mero de fluorocromos disponibles. | Puede detectar miles de genes simult√°neamente. |\n",
        "\n",
        "\n",
        "\n",
        "**Ejemplo:** En inmunolog√≠a, podemos usar citometr√≠a de flujo para contar y caracterizar c√©lulas T reguladoras en una muestra, pero con transcript√≥mica podemos analizar qu√© genes est√°n expresando, lo que nos permite entender mejor su funci√≥n y activaci√≥n.\n",
        "\n",
        ">\tEl an√°lisis transcript√≥mico complementa la citometr√≠a de flujo, pues **ayuda a dise√±ar mejores paneles de anticuerpos y marcadores üéØ.**\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## **1.3 - Herramientas Computacionales**\n",
        "\n",
        "Son programas, plataformas o **algoritmos dise√±ados para analizar, visualizar o interpretar datos biol√≥gicos complejos** (gen√©ticos, qu√≠micos, farmacol√≥gicos) de manera autom√°tica, r√°pida y precisa. \n",
        "\n",
        "**FUNCI√ìN:** simplificar tareas que manualmente ser√≠an imposibles o muy lentas.\n",
        "\n",
        "**CONSEJOS:**\n",
        "> Empieza con herramientas web (CIBERSORTx, STRING) antes de usar software local.\n",
        "\n",
        "> Usa datos de ejemplo: Muchas herramientas ofrecen datasets de prueba para practicar.\n",
        "\n",
        "> Busca tutoriales en YouTube: Hay gu√≠as visuales para cada herramienta.\n",
        "\n",
        "\n",
        "**EJEMPLOS:**\n",
        "\n",
        "> STRING (Web)\n",
        "Funci√≥n: Predecir interacciones entre prote√≠nas \n",
        "\n",
        "> CIBERSORTx (Web)\n",
        "Funci√≥n: Estimar proporciones de tipos celulares en una muestra\n",
        "\n",
        "> KEGG mapper (Web)\n",
        "Funci√≥n: Mapear genes o f√°rmacos en v√≠as bioqu√≠micas conocidas\n",
        "\n",
        "> hdWGCNA  (Software)\n",
        "Funci√≥n:  Encontrar grupos de genes que act√∫an juntos (m√≥dulos) en procesos biol√≥gicos."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "LWcVB8GoHApU"
      },
      "source": [
        "# **Secci√≥n 2: BioComputo** üß™üíª\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/Methodology.png\"\n",
        "        width=400\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## **2.1- Google Colaboratory**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/colab.jpg\"\n",
        "        width=225\n",
        "        height=125>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "#### Iniciaci√≥n de Google Drive"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NqNM4o4-Hugw",
        "outputId": "01b0dcfe-9924-4256-aacf-3f01d34e695f"
      },
      "outputs": [],
      "source": [
        "from google.colab import drive #Montar Google Drive\n",
        "drive.mount('/content/drive')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## **2.2 - Preprocesamiento de datos**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## 2.2.1 - scRNA-seq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Librerias\n",
        "\n",
        "import pandas as pd\n",
        "import numpy as np\n",
        "import umap\n",
        "import seaborn as sns\n",
        "import matplotlib.pyplot as plt\n",
        "from sklearn.preprocessing import StandardScaler"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "!pip install scanpy"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [],
      "source": [
        "import scanpy as sc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {},
      "outputs": [],
      "source": [
        "import anndata"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Cargar datos\n",
        "singlecell=pd.read_csv(\"C:/Users/Rocio/Desktop/singlecell_segmented.csv\", index_col=0, header=0) #Single Cell con 3,000 muestras celulares de 15 tipos celulares\n",
        "#---------- CAMBIAR RUTA ----------\n",
        "\n",
        "singlecell=singlecell.transpose()\n",
        "singlecell.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Separar la columna 'cell_types' de la matriz de expresi√≥n\n",
        "cell_types = singlecell['cell_types']  # Guardamos los tipos celulares\n",
        "singlecell = singlecell.drop(columns=['cell_types'])  # Quitamos la columna de la matriz "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Separar la fila \"feature_name\" de la matriz de expresi√≥n\n",
        "feature_name = singlecell.loc['feature_name'].values\n",
        "singlecell = singlecell.drop(index=\"feature_name\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Convertir valores de expresi√≥n a float\n",
        "singlecell = singlecell.astype(float)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## 2.2.2 - Visualizaci√≥n de Datos    "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Uso de Scanpy para el preprocesamiento \n",
        "\n",
        "En el an√°lisis de transcript√≥mica de c√©lulas individuales (single-cell RNA sequencing, scRNA-seq), trabajamos con grandes vol√∫menes de datos altamente dimensionales. Para poder extraer informaci√≥n √∫til, es necesario aplicar t√©cnicas de preprocesamiento, reducci√≥n de dimensionalidad, identificaci√≥n de grupos celulares y visualizaci√≥n.\n",
        "\n",
        "Scanpy es una de las herramientas m√°s eficientes y ampliamente utilizadas en Python para analizar datos de scRNA-seq. \n",
        "\n",
        "Pipeline:\n",
        "\n",
        " 1Ô∏è‚É£ Leer y organizar los datos de expresi√≥n \n",
        "\n",
        " 2Ô∏è‚É£ Filtrar c√©lulas y genes con baja expresi√≥n\n",
        "\n",
        " 3Ô∏è‚É£ Normalizar y aplicar log-transform\n",
        "\n",
        " 4Ô∏è‚É£ Identificar genes m√°s variables \n",
        "\n",
        " 5Ô∏è‚É£ Escalar datos de expresi√≥n \n",
        "\n",
        " 6Ô∏è‚É£ Construir representaciones de los datos con UMAP para visualizar los grupos celulares.\n",
        "\n",
        "\n",
        "En caso que posteriormente queramos usar R, Scanpy crea archivo compatibles con Seurat (R), para ellos construye un objeto AnnData que sigue el siguiente formato:"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/anndata_schema.svg\"\n",
        "        width=512\n",
        "        height=512>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Construir el objeto AnnData\n",
        "adata = anndata.AnnData(X=singlecell.values)\n",
        "\n",
        "# Asignar nombres correctos a filas y columnas\n",
        "adata.obs_names = singlecell.index  # C√©lulas\n",
        "adata.var_names = singlecell.columns  # Genes\n",
        "adata.obs[\"cell_types\"] = cell_types  # Agregar tipos celulares como metadatos\n",
        "adata.var[\"feature_name\"] = feature_name\n",
        "\n",
        "# Verificar estructura\n",
        "print(adata)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Filtrar c√©lulas y genes con baja expresi√≥n\n",
        "sc.pp.filter_cells(adata, min_genes=200)  # C√©lulas con al menos 200 genes detectados\n",
        "sc.pp.filter_genes(adata, min_cells=3)  # Genes expresados en al menos 3 c√©lulas\n",
        "adata\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Normalizar la expresi√≥n g√©nica (CPM-like normalization)\n",
        "sc.pp.normalize_total(adata, target_sum=1e4)\n",
        "\n",
        "# Aplicar log-transformaci√≥n\n",
        "sc.pp.log1p(adata)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "min_mean=0.0125 ‚Üí Excluye genes con una expresi√≥n promedio muy baja, porque suelen ser ruido.\n",
        "\n",
        "max_mean=3 ‚Üí Excluye genes con expresi√≥n excesivamente alta, que pueden ser housekeeping genes o artefactos.\n",
        "\n",
        "min_disp=0.5 ‚Üí Selecciona genes con alta dispersi√≥n (variabilidad relativa), lo que significa que su expresi√≥n var√≠a entre c√©lulas y no es uniforme."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Identificar los genes m√°s variables\n",
        "sc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)\n",
        "adata = adata[:, adata.var[\"highly_variable\"]]  # Filtrar solo genes variables\n",
        "adata"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {},
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "C:\\Users\\Rocio\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python310\\site-packages\\scanpy\\preprocessing\\_scale.py:317: UserWarning: Received a view of an AnnData. Making a copy.\n",
            "  view_to_actual(adata)\n"
          ]
        }
      ],
      "source": [
        "sc.pp.scale(adata, max_value=10)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "sc.pp.neighbors(adata, n_neighbors=30, metric=\"euclidean\")\n",
        "sc.tl.umap(adata, min_dist=0.6)\n",
        "sc.pl.umap(adata, color=\"cell_types\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Extraer la matriz de expresi√≥n desde adata.X y convertirla en DataFrame\n",
        "df_expression = pd.DataFrame(\n",
        "    adata.X.T,  # Transponemos para que los genes sean filas y las c√©lulas columnas\n",
        "    index=adata.var[\"feature_name\"],  # Nombres de los genes como filas\n",
        "    columns=adata.obs_names  # Nombres de las c√©lulas como columnas\n",
        ")\n",
        "df_expression.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 25,
      "metadata": {},
      "outputs": [],
      "source": [
        "df_expression.to_csv(\"C:/Users/Rocio/Desktop/matriz_expresion_preprocesada.txt\", sep=\"\\t\", index=True, header=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {},
      "outputs": [],
      "source": [
        "# Guardar como un .h5ad para despu√©s convertirlo en .rds en R (opcional)\n",
        "adata.write(\"C:/Users/Rocio/Desktop/matriz_expresion_preprocesada.h5ad\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## 2.2.3 - RNA-seq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Lectura de datos"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5EqVXG_Wbb24"
      },
      "source": [
        "### Procesamiento de datos RNA-seq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 517
        },
        "id": "gJAKkGb6bkOz",
        "outputId": "1b470a7a-c07d-40e0-b7b2-fc0456eea898"
      },
      "outputs": [],
      "source": [
        "# Carga de datos\n",
        "rnaseq=pd.read_csv(\"/Users/gusano2398gmail.com/Desktop/Prueba/RNA-seq.csv\") # Direcci√≥n en la que se encuentra el archivo RNA-seq\n",
        "#---------- CAMBIAR RUTA ----------\n",
        "\n",
        "print(\"Base de datos \\n\")\n",
        "\n",
        "rnaseq = rnaseq.rename({'gene_symbol': 'Brain Samples', 'b': 'Y'}, axis=1) # Cambiamos de nombre una columna\n",
        "rnaseq.head() # Mostrar los datos en forma de tabla solo los primeros 5 elementos\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Filtrado de datos"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5YP3t7FobucL",
        "outputId": "706491e2-178d-4e79-d10f-ec1201b2ebd8"
      },
      "outputs": [],
      "source": [
        "ages = [\"8 pcw\", \"40 yrs\"] #Edades en las que quieres inferir la proporci√≥n celular\n",
        "\n",
        "\n",
        "# Filtara el DF\n",
        "filtered_df = rnaseq[rnaseq[\"Age\"].isin(ages)]\n",
        "\n",
        "# Eliminar columna de edad\n",
        "filtered_df=filtered_df.drop(\"Age\", axis=1)\n",
        "\n",
        "# Mostrar los resultados\n",
        "filtered_df=filtered_df.set_index([\"Brain Samples\"])\n",
        "print(\"Base de datos con Indice\\n\")\n",
        "print(filtered_df)\n",
        "\n",
        "filtered_df\n",
        "\n",
        "# Define the Google Drive path\n",
        "save_path = \"/Users/gusano2398gmail.com/Desktop/Prueba/\"\n",
        "\n",
        "#---------- CAMBIAR RUTA ----------\n",
        "\n",
        "#Exportar los datos a un archivo tipo csv\n",
        "filtered_df.to_csv(save_path+\"RNA-seq-Filtered.csv\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Guardar datos en formato txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uC92ngHocM57",
        "outputId": "86b445d1-d786-4efa-b4ef-7349c179c088"
      },
      "outputs": [],
      "source": [
        "# Carga de datos previamente segmentados\n",
        "rnaseq=pd.read_csv(save_path+\"RNA-seq-Filtered.csv\", header=None)\n",
        "print(\"Base de datos \\n\")\n",
        "rnaseq=rnaseq.transpose()\n",
        "# Save the file\n",
        "np.savetxt(save_path+\"Bulk.txt\", rnaseq, fmt='%s', delimiter='\\t')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Visualizaci√≥n de datos segmentados finales"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 650
        },
        "id": "_Nz10GQpnwBW",
        "outputId": "14613dc8-468e-46ba-e8a2-93dda4a52b61"
      },
      "outputs": [],
      "source": [
        "rnaseq=pd.read_csv(save_path+\"Bulk.txt\", header=None, delimiter=\"\\t\", index_col=0)\n",
        "rnaseq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## **2.3 - CIBERSORTx** üìä"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/Methodology.png\"\n",
        "        width=400\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qAopxgDgczze"
      },
      "source": [
        "### Link a CIBERSORTx: https://cibersortx.stanford.edu/\n",
        "\n",
        "Pasos a seguir\n",
        "\n",
        "\n",
        "1.   Crear cuenta\n",
        "2.   Iniciar ses√≠on\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## 2.3.1 - Firmas gen√©ticas"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/Firmas.png\"\n",
        "        width=800\n",
        "        height=600>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## 2.3.2 - Inferencia Celular"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/Proporcion.png\"\n",
        "        width=800\n",
        "        height=600>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/CIBERSORTx.png\"\n",
        "        width=700\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## **2.4 - hdWGCNA** üï∏Ô∏è"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "hdWGCNA es una herramienta bioinform√°tica utilizada aplicada en an√°lisis de redes de coexpresi√≥n g√©nica en estudios de transcript√≥mica que nos permite:\n",
        "\n",
        "- **Identificar grupos de genes que trabajan juntos en diferentes condiciones**, como en enfermedades o en respuesta a f√°rmacos\n",
        "\n",
        "- **Encontrar biomarcadores y** posibles** dianas terap√©uticas** analizando qu√© genes est√°n m√°s conectados en una red\n",
        "\n",
        "- Comprender c√≥mo un f√°rmaco afecta la expresi√≥n g√©nica y predecir posibles efectos adversos\n",
        "\n",
        "- Estudiar enfermedades complejas como el c√°ncer o trastornos neurol√≥gicos a nivel molecular\n",
        "\n",
        "\n",
        "Website: https://smorabit.github.io/hdWGCNA/  "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/Module.png\"\n",
        "        width=400\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/ModuleCoexpression.png\"\n",
        "        width=400\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## **2.5 - STRING**\n",
        "\n",
        "**STRING** (Search Tool for Retrieval of Interacting Genes/Proteins) es una plataforma web gratuita que **permite estudiar interacciones funcionales entre prote√≠nas, genes o compuestos qu√≠micos**. Combina datos experimentales, predicciones computacionales y miner√≠a de textos para construir redes biol√≥gicas.\n",
        "\n",
        "**Enlace: https://string-db.org**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/STRING.png\"\n",
        "        width=900\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### An√°lisis avanzado (sin programar)\n",
        "\n",
        "<fig>\n",
        "<img    src=\"Edits/stringview.png\"\n",
        "        width=900\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## **2.6- STITCH** \n",
        "\n",
        "STITCH es una base de datos de **interacciones prote√≠na-qu√≠micos** que integra diversas fuentes de evidencia experimental y que permite seleccionar manualmente con informaci√≥n de miner√≠a de texto y predicciones de interacci√≥n.\n",
        "\n",
        "\n",
        "<fig>\n",
        "<img    src=\"Edits/STITCH.png\"\n",
        "        width=900\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>\n",
        "\n",
        "Website: http://stitch.embl.de/ "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# **Secci√≥n 3: Ciencia de Datos** üß¨üíª"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "K89uxZNUdVnh"
      },
      "source": [
        "## 5.1- Interpretaci√≥n de Resultados\n",
        "\n",
        "<fig>\n",
        "<img    src=\"Edits/Visualization.png\"\n",
        "        width=650\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>\n",
        "\n",
        ".\n",
        ".\n",
        ".\n",
        ".\n",
        "\n",
        "La interpretaci√≥n de resultados en biolog√≠a computacional es el proceso de traducir datos num√©ricos o estad√≠sticos en conocimiento biol√≥gicamente relevante, vincul√°ndolos con mecanismos celulares, enfermedades o aplicaciones farmac√©uticas. Se realiza en varias etapas:\n",
        "\n",
        "**> Validaci√≥n t√©cnica:**\n",
        "\n",
        "Confirmar que los resultados son robustos (ej.: repetir an√°lisis con distintos par√°metros o subconjuntos de datos).\n",
        "\n",
        "\n",
        "**> Contextualizaci√≥n biol√≥gica:**\n",
        "\n",
        "Relacionar genes, prote√≠nas o v√≠as identificadas con funciones conocidas usando bases de datos como KEGG, GO o Reactome.\n",
        "\n",
        "**> Integraci√≥n con literatura:**\n",
        "\n",
        "Comparar resultados con estudios previos\n",
        "\n",
        "**> Herramientas de apoyo:**\n",
        "\n",
        "**Anotaci√≥n funcional:** DAVID, Enrichr para identificar v√≠as enriquecidas.\n",
        "\n",
        "**Redes biol√≥gicas:** Cytoscape para visualizar interacciones entre genes/prote√≠nas descubiertos.\n",
        "\n",
        "**Plataformas farmacol√≥gicas:** DrugBank para cruzar dianas identificadas con f√°rmacos existentes.\n",
        "\n",
        "**> Traducci√≥n a aplicaciones pr√°cticas:**\n",
        "\n",
        "Generar hip√≥tesis experimentales\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "*¬øQu√© significa esto para la enfermedad o el f√°rmaco que estudio?*"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/figure.png\"\n",
        "        width=500\n",
        "        height=700>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/Trends.png\"\n",
        "        width=700\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/layers_down.png\"\n",
        "        width=700\n",
        "        height=600>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/hdWGCNA_string.png\"\n",
        "        width=1020\n",
        "        height=550>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# **Laboratorio de Innovaci√≥n Biodigital**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "<fig>\n",
        "<img    src=\"Edits/lab.jpg\"\n",
        "        width=600\n",
        "        height=400>\n",
        "<figcaption>\n",
        "</figure>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "**Algunos de los temas y herramientas que trabajamos en el laboratorio incluyen:**\n",
        "\n",
        "**Herramientas**\n",
        "- üîπ **Programaci√≥n en R y Python** - Visualizaci√≥n de datos y an√°lisis\n",
        "- üîπ **hdWGCNA** ‚Äì An√°lisis de redes de coexpresi√≥n gen√©tica\n",
        "- üîπ **Panther y STRING** ‚Äì Herramientas para el an√°lisis de prote√≠nas y la identificaci√≥n de interacciones moleculares.\n",
        "- üîπ **AlphaFold y RosettaFold** ‚Äì Modelado de estructuras de prote√≠nas para estudiar sus funciones biol√≥gicas.\n",
        "- üîπ **IAs: ML y DL** ‚Äì Predicci√≥n de patrones en grandes datasets\n",
        "\n",
        "---\n",
        "\n",
        "\n",
        "\n",
        "Si est√°n interesados en **colaborar con nosotros** o en aprender m√°s sobre **an√°lisis de datos biom√©dicos avanzados**, estamos **ofreciendo cursos especializados** en estos temas. Adem√°s, si su laboratorio o equipo tiene proyectos en los que podamos aportar con nuestras herramientas y conocimientos, no duden en ponerse en contacto.\n",
        "\n",
        "\n",
        "Para m√°s informaci√≥n o consultas, no duden en contactarnos. "
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.9.5"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
